builder/cloudstack/step_create_instance.go

```diff
@@ -92,7 +96,9 @@ func (s *stepCreateInstance) Run(state multistep.StateBag) multistep.StepAction
 
 		ud, err := s.generateUserData(config.UserData, config.HTTPGetOnly)
 		if err != nil {
+			err := fmt.Errorf("Failed to interpolate user_data: %s", err)
```

Comment:
The `generateUserData` function already returns well formatted and detailed errors, so we should not try to add a general annotation here (the function does more then interpolation).